plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/InternalHiveSplitFactory.java

```diff
@@ -203,7 +203,7 @@ public String getPartitionName()
 
         int statementId = 0;
 
-        if (transaction.isDelete()) {
+        if (transaction.isDelete() || transaction.isUpdate()) {
```

Comment:
Thanks for adding explanation https://github.com/trinodb/trino/issues/7186#issuecomment-802439753

> * Fix allocation of statement IDs for ACID tables, which could result in query failure or
  data loss due to creating multiple delta files with the same name. ({issue}`7302`)

Sounds like this should be tested.